util/qc_executor.py

```diff
@@ -84,9 +98,24 @@ def qc_check(self, parameter, dataset):
             if 'strict_validation' not in qcs[function_name]:
                 local_qc_args[function_name]['strict_validation'] = False
 
+            qc_process = None
             try:
-                module = importlib.import_module(ParameterFunction.query.filter_by(function=function_name).first().owner)
-                results = getattr(module, function_name)(**local_qc_args.get(function_name))
+                # call qc function in a separate process to deal with crashes, e.g. segfaults
+                module = importlib.import_module(ParameterFunction.query.filter_by(function=function_name)
+                                                 .first().owner)
+                parent_conn, child_conn = Pipe()
+                qc_process = Process(target=self.qc_process_wrapper, args=(child_conn, getattr(module, function_name),
+                                                                           local_qc_args.get(function_name)))
+                qc_process.start()
+                # close unused end of duplex pipe
+                child_conn.close()
```

Comment:
Isn't it enough to close child_conn in the qc_process_wrapper function?

Reply:
No. It's confusing. See the quote from stackoverflow below:
"You should be closing pipe ends in processes that don't need them. The issue is that once you launch the subprocess, the kernel is tracking two open references to the send connection of the pipe: one in the parent process, one in your subprocess.
The send connection object is only being closed in your subprocess, leaving it open in the parent process, so your conn.recv() call won't raise EOFError. The pipe is still open."